Week-6/tictactoe/tictactoe.js

```diff
@@ -0,0 +1,140 @@
+let playerTurn = true;
+let computerMoveTimeout = 0;
+
+const gameStatus = {
+  MORE_MOVES_LEFT: 1,
+  HUMAN_WINS: 2,
+  COMPUTER_WINS: 3,
+  DRAW_GAME: 4,
+};
+
+window.addEventListener("DOMContentLoaded", domLoaded);
+
+function domLoaded() {
+  // Setup the click event for the "New game" button
+  const newBtn = document.getElementById("newGameButton");
+  newBtn.addEventListener("click", newGame);
+
+  // Create click-event handlers for each game board button
+  const buttons = getGameBoardButtons();
+  for (let button of buttons) {
+    button.addEventListener("click", function () {
+      boardButtonClicked(button);
+    });
+  }
+
+  // Clear the board
+  newGame();
```

Comment:
There's a risk of a memory leak with computerMoveTimeout if newGame is called before the timeout is completed. Good to see it being cleared on new game starts, just make sure it's consistently handled on all relevant events.

Centralize timeout management
```
function setComputerMoveTimeout() {
  clearComputerMoveTimeout(); // Clear any existing timeout
  computerMoveTimeout = setTimeout(makeComputerMove, 1000);
}
```

```
function clearComputerMoveTimeout() {
  if (computerMoveTimeout) {
    clearTimeout(computerMoveTimeout);
    computerMoveTimeout = 0;
  }
}
```


Update within switchTurn function
```
function switchTurn() {
  // ... existing code ...
  if (!playerTurn) {
    setComputerMoveTimeout();
  }
  // ... existing code ...
}
```

Update within newGame function
```
function newGame() {
  clearComputerMoveTimeout();
  // ... remaining logic for newGame ...
}
```